File: healthmate-frontend/src/services/dietPlan.service.js
import api from './api';

const createDietPlan = (data) => {
    return api.post('/diet-plans', data);
};

const getDietPlansByTrainer = (trainerId) => {
    return api.get(`/diet-plans/trainer/${trainerId}`);
};

const deleteDietPlan = (id) => {
    return api.delete(`/diet-plans/${id}`);
};

const dietPlanService = {
    createDietPlan,
    getDietPlansByTrainer,
    deleteDietPlan
};

export default dietPlanService;

File: healthmate-backend/src/main/java/com/healthmate/service/DietPlanService.java
package com.healthmate.service;

import com.healthmate.model.DietPlan;
import com.healthmate.repository.DietPlanRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class DietPlanService {
    @Autowired
    private DietPlanRepository dietPlanRepository;

    public DietPlan createDietPlan(DietPlan dietPlan) {
        return dietPlanRepository.save(dietPlan);
    }

    public List<DietPlan> getDietPlansByTrainer(String trainerId) {
        return dietPlanRepository.findByTrainerId(trainerId);
    }

    public void deleteDietPlan(String id) {
        dietPlanRepository.deleteById(id);
    }
}
